### PR TITLE
build(travis): Split webpack + tsc from test suite for frontend

### PR DESCRIPTION
**Pull request by billyvg**
### Changes
* For the frontend job in CI, split `webpack` off from the test suite job and also run `tsc` in CI.
* Removes `ts-loader` as it is a bit redundant since `babel-loader` can handle typescript files now.
  * This means that our cloud build will *NOT* include type checks, @BYK is this fine, or should we include a call to `tsc` before building?

### CODE REVIEW

**Comment by billyvg** [comment]
I don't know if we actually want this yet

File: .travis.yml [hunk 1]
```diff
@@ -187,10 +187,18 @@ matrix:
       env: TEST_SUITE=plugins DB=postgres PERCY_TOKEN=${PLUGIN_PERCY_TOKEN}
 
     - python: 2.7
-      name: 'Frontend'
+      name: 'Frontend [test]'
       env: TEST_SUITE=js
       before_install:
-        - *base_install
```
Comment (billyvg) [inline .travis.yml]: Does our frontend test suite actually need this? It has no python/pip deps. /cc @joshuarli

File: webpack.config.js [hunk 2]
```diff
@@ -237,19 +233,11 @@ let appConfig = {
...
         use: babelLoaderConfig,
       },
-      {
-        test: /\.tsx?$/,
-        include: [staticPrefix],
-        exclude: /(vendor|node_modules|dist)/,
-        // Make sure we typecheck in CI, but not for local dev since that is run with
-        // the fork-ts plugin
-        use: SHOULD_FORK_TS ? babelLoaderConfig : [babelLoaderConfig, tsLoaderConfig],
-      },
```
Comment (billyvg) [inline webpack.config.js]: This requires GS changes.

File: .travis.yml (hunk 1 above)
Comment (joshuarli) [reply .travis.yml]: Doesn't look like it needs it to me. Should work.

File: webpack.config.js (hunk 2 above)
Comment (billyvg) [reply webpack.config.js]: https://github.com/getsentry/getsentry/pull/3704

**Review by evanpurkhiser — APPROVED** [review]

**Comment by BYK** [comment]
I like splitting the test suite from Webpack but not sure if separating the `tsc` run should be a part of this change. Can you help me understand why these are coupled and what the benefits are?

**Comment by billyvg** [comment]
@BYK are your concerns more regarding the scope of the PR, or about removing `ts-loader` itself? I can split this PR if that's what you're asking for, but I thought these were closely related enough to just bundle them together. I would actually just move the travis changes into a new PR.

As for the reasoning behind removing `ts-loader` - it's a dependency that adds typechecking into the webpack toolchain, but it also complicates our config a bit, i.e. for tsx files, we run `babel-loader` *and* `ts-loader`, but only on CI.

We were also seeing issues with `ts-loader` while trying to add typescript support in `getsentry`.

**Comment by BYK** [comment]
@billyvg - I preferred if the PR was split into 2 for safety reasons but at this point I see little benefit of splitting which would also delay the merging even more.

Regarding `ts-loader`, my main concern was removing type checking from the `build_assets` step even though it runs on CI. Do you think you can add it there so the whole wheel/docker build fails on TypeScript issues? I'm guessing this will come with a minor performance hit.

**Comment by billyvg** [comment]
@BYK I'll include `tsc` when we build static assets -- perf-wise this may be an improvement, at the very least it'll be similar since it already was running type-checks with some webpack overhead.

**Pull request merged**